common/src/main/java/org/apache/nemo/common/punctuation/Latencymark.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.nemo.common.punctuation;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Latency mark is conveyor that has data for debugging.
+ * It is created only from source vertex and record the timestamp when it is created and taskId where it is created.
```

Comment:
I considered it is possible that there are multiple stage that has source vertex. So I added createdTaskId field.
latencymark does not record only latency between source to sink. It record latency between every task in the path. So I added lastTaskId field to track path.

Reply:
First of all, Pair of createdTaskId and timestamp indicates the starting point. For examples,
      -----------B-----------
    /                                        \          A to E indicates tasks
 A ------------C-----------E        dash indicates edges
   \                                         /           A is source, and E is sink
     ------------D----------     
Latencymark is created from task A with timestamp and createdTaskId. 
And the lastTaskId needs to follow each path. 
When Latencymark reached at task E, there are totally three path, A-B-E, A-C-E, A-D-E. And I want to 
know latency for each path. So I make it hold the previous task Id. So when Latencymark reached at task E, Task E received three latencymark. And the lastTaskId will be B, C, and E, respectively. Of course, The lastTaskId itself does not have a meaning, but I expected that we would know the latency for each path if we followed the lastTaskId. This part may be controversial and I think it will be okay to remove if it is deemed unnecessary.

Reply:
What happen in the following situation?
-----------B ------ C --------
/                                 \                  \
A -------- D ------   E -------- F

There are three paths: 
1) A->B->C->F 
2) A->B->C->E->F
3) A->D->E->F

How can we distinguish 2) and 3) ?

Reply:
traverse time from E to F will not significantly different. So if we want to distinguish 2) and 3) we can check the record of the previous task. Then we can find a difference between C to E and D to E. Rather than the metric itself is meaningful, It needs an effort to gather transmission times between nodes

I think the departure time from the previous task will be important. But In the current structure, There seems little bit difficult to judge. because several Latencymark with the same source task id and created time can be sent. 

I will change to record the time when the latencymark departed from the previous task or modify to send only unique(with unique pair of createTaskid and createdTimestamp) Latencymark